src/tracepoint_format_parser.cpp

```diff
@@ -23,11 +29,14 @@ bool TracepointFormatParser::parse_wildcard_ap(ast::Probe *probe,
   std::string format_file_path = "/sys/kernel/debug/tracing/events/" +
                                  category + "/" + event_name + "/format";
 
-  glob_t glob_result;
+  // Use a unique_ptr to have to system call globfree() for us instead of having
+  // to do it ourselves. The braces after `new glob_t()` are required to init
+  // the struct. `globfree()` on random data is unsafe
+  std::unique_ptr<glob_t, decltype(&freeglob)> glob_result(new glob_t(),
```

Comment:
Custom deleter on a unique_ptr works, but in the past I've used the scope guard pattern (it works well and is really resuable).

Source: https://github.com/facebookincubator/oomd/blob/master/src/oomd/util/ScopeGuard.h
Example: https://github.com/facebookincubator/oomd/blob/7bffe9e0d4143ca8acacc4bc640888d6e14e36b4/src/oomd/util/ScopeGuardTest.cpp#L27-L37

Maybe copy paste that into bpftrace?

Reply:
I guess we can give that a go :)